### ISSUE TITLE
Error in difference function - 2.1.0

### ISSUE THREAD

**Issue by ranajit484** [orientechnologies/orientdb#4832]
Hi,
There seems to be still an error in difference() function in the version 2.1.0.

We are executing following set of queries:
```
CREATE class roles
CREATE property roles.ROLE string
CREATE property roles.group_id INTEGER
INSERT INTO roles (ROLE, group_id) VALUES ('CREATE_JOB', 1)
INSERT INTO roles (ROLE, group_id) VALUES ('CREATE_JOB', 2)
INSERT INTO roles (ROLE, group_id) VALUES ('UPDATE_JOB', 1)
INSERT INTO roles (ROLE, group_id) VALUES ('UPDATE_JOB', 2)
INSERT INTO roles (ROLE, group_id) VALUES ('VIEW_ALL_JOBS', 2)
```
And after that we are running following query:

```
SELECT expand( $3 ) LET $1 = ( SELECT ROLE FROM roles WHERE group_id=1), $2 = ( SELECT ROLE FROM  roles WHERE group_id=2), $3 = difference( $1, $2 )
```
The output should be an empty result set.
The output in version 2.0.8: empty result set (which is correct)
The output in version 2.1.0: 

![alt text](https://sc-cdn.scaleengine.net/i/4154c601c999cf3481a7201478cdd630.png)

And for following query:

```
SELECT expand( $3 ) LET $1 = ( SELECT ROLE FROM roles WHERE group_id=1), $2 = ( SELECT ROLE FROM  roles WHERE group_id=2), $3 = difference( $2, $1 )
```

the output should return single row i.e. VIEW_ALL_JOBS
The output in version 2.0.8: VIEW_ALL_JOBS (which is correct)
The output in version 2.1.0: 

![alt text](https://sc-cdn.scaleengine.net/i/1efb005c84e2fa7f54b38e323df2c99d.png)

We have tried this in version 2.1-alpha as well. Is this the expected behavior for 2.1.0 version?

Similar issue can be found here which was for 2.1rc3:
[#4366](https://github.com/orientechnologies/orientdb/issues/4366)

**Comment by lvca** [comment]
This is documented in the upgrade guide: http://orientdb.com/docs/last/Release-2.1.0.html#difference-function

**Comment by ranajit484** [comment]
@lvca I don't think this issue has been fixed. As I tested above example in version 2.1.0 as well as in 2.2-snapshot and according to this [http://orientdb.com/docs/last/Release-2.1.0.html#difference-function] (http://orientdb.com/docs/last/Release-2.1.0.html#difference-function) , when we try with symmetric difference function, it is throwing an error saying 'No function with name 'symmetricdifference''. 

Also, that would be very helpful if you provide any simple working  examples in the documentation for these functions (difference, symmetricdifference), as currently the examples provided are seems to be less informative.

Thanks.
